record/record.go

```diff
@@ -24,11 +24,12 @@
 //		records := record.NewReader(r)
 //		for {
 //			rec, err := records.Next()
-//			if err == io.EOF {
+//			if err == io.EOF || err == io.ErrUnexpectedEOF {
```

Comment:
You return io.ErrUnexpectedEOF when the previous read did not fill the buffer, and we haven't consumed all of the available data. This shouldn't really ever happen. I'm not sure Recover'ing from this this point makes sense.